### PR TITLE
Updates in IT

### PR DESCRIPTION
**Pull request by himanshu-kun**
**What this PR does / why we need it**:

Integration Tests have been updated.

Updates:

* Taint is added to any node present before the test suite starts and after the test suite finishes (successfully or unsuccessfully)
* `--ignore-taint` flag has been updated in Makefile, for `make start` makerule.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other developer
Nodes present in the cluster prior to IT would now be tainted with taint `testing.node.gardener.cloud/initial-node-blocked`. This taint is removed once IT finishes (pass or fail)
```

### CODE REVIEW

File: cluster-autoscaler/integration/framework.go [hunk 1]
```diff
@@ -54,6 +55,28 @@ func rotateLogFile(fileName string) (*os.File, error) {
 	return os.Create(fileName)
 }
 
+func (driver *Driver) taintOrUnTaintInitialNodes(toTaint bool) error {
```
Comment (rishabh-11) [inline cluster-autoscaler/integration/framework.go]: Shouldn't we have different methods for adding and removing taints? That way, we don't have to pass this boolean to determine if we are adding or removing them.

File: cluster-autoscaler/integration/framework.go [hunk 2]
```diff
@@ -315,39 +338,84 @@ func getDeploymentObject(replicas int32, resourceCpu string, resourceMemory stri
...
 }
 
-func (driver *Driver) deployWorkload(replicas int32, workloadName string) error {
-	deployment := getDeploymentObject(replicas, fmt.Sprint(cpuResource-1), mediumMemory, workloadName)
+func (driver *Driver) deployWorkload(replicas int32, workloadName string, canTolerateTaintPlacedOnInitialNodes bool) error {
+	// TODO(himanshu-kun): Remove such a dependency on approximating system components space . This changes over time.
+	assumedSystemComponentsUsedSpace := *resource.NewMilliQuantity(1000, resource.DecimalSI)
+	cpuRequested := cpuResource.DeepCopy()
+	cpuRequested.Sub(assumedSystemComponentsUsedSpace)
+	tolerationsToInitialNodeTaint := []v1.Toleration{
```
Comment (rishabh-11) [inline cluster-autoscaler/integration/framework.go]: Can we have this as a separate variable? The same variable is used in other methods as well

File: cluster-autoscaler/integration/framework.go (hunk 1 above)
Comment (himanshu-kun) [reply cluster-autoscaler/integration/framework.go]: okay I'll add separate functions

File: cluster-autoscaler/integration/framework.go (hunk 2 above)
Comment (himanshu-kun) [reply cluster-autoscaler/integration/framework.go]: addressed in [a8a7cef](https://github.com/gardener/autoscaler/pull/187/commits/a8a7cef0877b589973075d0e53b70f2974810ab9)

File: cluster-autoscaler/integration/framework.go (hunk 1 above)
Comment (himanshu-kun) [reply cluster-autoscaler/integration/framework.go]: addressed in [a8a7cef](https://github.com/gardener/autoscaler/pull/187/commits/a8a7cef0877b589973075d0e53b70f2974810ab9)

**Review by rishabh-11 — APPROVED** [review]
Thanks a lot for the changes.
/lgtm

**Pull request merged**